lib/runtime-core/src/codegen.rs

```diff
@@ -112,14 +112,45 @@ pub trait ModuleCodeGenerator<FCG: FunctionCodeGenerator<E>, RM: RunnableModule,
     /// Checks the precondition for a module.
     fn check_precondition(&mut self, module_info: &ModuleInfo) -> Result<(), E>;
     /// Creates a new function and returns the function-scope code generator for it.
-    fn next_function(&mut self, module_info: Arc<RwLock<ModuleInfo>>) -> Result<&mut FCG, E>;
+    fn next_function(
+        &mut self,
+        module_info: Arc<RwLock<ModuleInfo>>,
+        loc: (u32, u32),
+    ) -> Result<&mut FCG, E>;
     /// Finalizes this module.
-    fn finalize(self, module_info: &ModuleInfo) -> Result<(RM, Box<dyn CacheGen>), E>;
+    fn finalize(
+        self,
+        module_info: &ModuleInfo,
+    ) -> Result<((RM, Option<DebugMetadata>), Box<dyn CacheGen>), E>;
```

Comment:
Why did you do that?

Shouldn't be simpler a single tuple?

```suggestion
    ) -> Result<(RM, Option<DebugMetadata>, Box<dyn CacheGen>), E>;
```